apps/server/src/modules/user-login-migration/service/common-user-login-migration.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { Permission, SchoolDO, User, UserLoginMigrationDO } from '@shared/domain';
+import { AuthorizationContext, AuthorizationContextBuilder, AuthorizationService } from '@src/modules/authorization';
+import { SchoolService } from '@src/modules/school';
+import { UserLoginMigrationService } from './user-login-migration.service';
+
+@Injectable()
+export class CommonUserLoginMigrationService {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly userLoginMigrationService: UserLoginMigrationService,
+		private readonly schoolService: SchoolService
+	) {}
+
+	async ensurePermission(userId: string, schoolId: string): Promise<void> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		const school: SchoolDO = await this.schoolService.getSchoolById(schoolId);
+		const context: AuthorizationContext = AuthorizationContextBuilder.write([Permission.USER_LOGIN_MIGRATION_ADMIN]);
+
+		this.authorizationService.checkPermission(user, school, context);
```

Comment:
@alweber-cap you want to speak about it in arc chapter again. But the authorisation is part of the UC for now. I approve a lot of PRs that do it in other way. But if we not change our strict way, your team run into a situation that it must refactore all the services and tests. This is something that you are currently responsible for.